core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -15,7 +15,67 @@ import scala.concurrent.duration._
 
 /**
   * A stream producing output of type `O` and which may evaluate `F`
-  * effects. If `F` is [[Pure]], the stream evaluates no effects.
+  * effects.
+  * Or: a lazy sequence of computations that may evaluate `F` effects `F` and yield values of type `O`.
+  *
+  * Streams are purely functional: a `Stream[F, O]` is a description for a computation,
+  * much like `cats.effect.IO` or `scalaz.Task` are just descriptions of computations.
```

Comment:
I'd replace the reference to `scalaz.Task` with Monix and ZIO perhaps. The old scalaz task was `scalaz.concurrent.Task` but that's ancient history now.